lib/workmanager.dart

```diff
@@ -235,15 +236,28 @@ class Workmanager {
   }
 
   /// Cancels a task by its [uniqueName]
-  static Future<void> cancelByUniqueName(final String uniqueName) async =>
-      await _foregroundChannel
-          .invokeMethod("cancelTaskByUniqueName", {"uniqueName": uniqueName});
+  static Future<void> cancelByUniqueName(final String uniqueName) async {
+    _assertNotIOS();
+    await _foregroundChannel
+        .invokeMethod("cancelTaskByUniqueName", {"uniqueName": uniqueName});
+  }
 
   /// Cancels a task by its [tag]
-  static Future<void> cancelByTag(final String tag) async =>
-      await _foregroundChannel.invokeMethod("cancelTaskByTag", {"tag": tag});
+  static Future<void> cancelByTag(final String tag) async {
+    _assertNotIOS();
+    await _foregroundChannel.invokeMethod("cancelTaskByTag", {"tag": tag});
+  }
 
   /// Cancels all tasks
-  static Future<void> cancelAll() async =>
-      await _foregroundChannel.invokeMethod("cancelAll");
+  static Future<void> cancelAll() async {
+    _assertNotIOS();
+    await _foregroundChannel.invokeMethod("cancelAll");
+  }
+
+  static void _assertNotIOS() {
+    if (Platform.isIOS) {
```

Comment:
How about assert(Platform.isIOS) ?

Reply:
To clarify, I was thinking of this for 2 reasons.
* Asserting that we are on Android makes more sense.
* Assert only evaluates/throws in debug

Reply:
Yes, I think I made a boo boo there it should either be `!Platform.isIOS` or `Platform.isAndroid`.
Question remains, is an AssertionError enough in debug or do we want to throw an Exception?

Reply:
AssertionError sounds good to me. It's commonly used in the Flutter framework for 'illegal states'.